infra/discovery/src/lib/search.js

```diff
@@ -220,20 +220,70 @@ class Listing {
       esQuery.bool.filter.push(innerFilter)
     })
 
-    /* When users boost their listing using OGN tokens we boost that listing in elasticSearch.
-     * For more details see document: https://docs.google.com/spreadsheets/d/1bgBlwWvYL7kgAb8aUH4cwDtTHQuFThQ4BCp870O-zEs/edit#gid=0
-     *
-     * If boost slider's max value (currently 100) on listing-create changes this factor needs tweaking as well. (see referenced document)
-     */
-    const boostScoreQuery = {
+    const scoreQuery = {
       function_score: {
         query: esQuery,
-        field_value_factor: {
-          field: 'commissionPerUnit',
-          factor: 0.0000000000000000005, // the same as delimited by 20
-          missing: 0
-        },
-        boost_mode: 'sum'
+
+        script_score: {
+          script: {
+            params: {
+              // Strongly reccomended to pass date in as a paramter since:
+              // - All nodes in an elastic search cluster will be using the same value
+              // - Script itself stays the same, and never needs to be recompiled
+              now: new Date().getTime()
+            },
+            source: `double score = _score;
+
+            // Downrank "cheap" listings to hide wash/spam transactions.
+            // We might want to make this a smoother transition later.
+            // ETH value may need to become a dynamic parameter later.
+            if (doc['price.amount'] != null && doc['price.currency.id'] != null) {
+              if(doc['price.currency.id'].value == "fiat-USD"){
```

Comment:
How about the other currencies we currently support ? They are not super popular (see stats below) but might be worth adding them for future ?

origin=> select data->'price'->'currency'->'id', count(*) from listing group by data->'price'->'currency'->'id' order by count(*) desc;
  ?column?   | count 
-------------+-------
 "token-ETH" |  4730
 "fiat-USD"  |  4272
 "fiat-EUR"  |    84
 "fiat-KRW"  |    10
 "fiat-GBP"  |     6
 "fiat-JPY"  |     4
 "fiat-CNY"  |     4

Reply:
I've added Euros. Looks like this takes care of 99% of the listings.